include/rocksdb/sst_file_reader.h

```diff
@@ -0,0 +1,47 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#ifndef ROCKSDB_LITE
+
+#include "rocksdb/slice.h"
+#include "rocksdb/options.h"
+#include "rocksdb/iterator.h"
+#include "rocksdb/table_properties.h"
+
+namespace rocksdb {
+
+// SstFileReader is used to read sst files that are generated by DB or
+// SstFileWriter.
+class SstFileReader {
+ public:
+  SstFileReader(const Options& options);
+
+  ~SstFileReader();
+
+  // Prepares to read from the file located at "file_path".
+  Status Open(const std::string& file_path);
+
+  // Gets a value with the key from the table.
+  Status Get(const ReadOptions& options,
+             const Slice& key, PinnableSlice* value);
+
+  // Returns a new iterator over the table contents.
+  Iterator* NewIterator(const ReadOptions& options);
```

Comment:
maybe document how it returns only latest keys unless `ReadOptions::snapshot` is set to a live snapshot

Do you plan to use this API for reading older snapshots? If so is this API easier for you compared to having an iterator that returns internal keys?